src/SprykerSdk/Acp/Schema/SchemaExtender.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Acp\Schema;
+
+use cebe\openapi\Reader;
+use cebe\openapi\spec\OpenApi;
+use Symfony\Component\Process\Process;
+use Transfer\CreateDefaultEndpointsRequestTransfer;
+use Transfer\CreateDefaultEndpointsResponseTransfer;
+use Transfer\MessageTransfer;
+
+class SchemaExtender implements SchemaExtenderInterface
+{
+    /**
+     * @var \SprykerSdk\Acp\Schema\SchemaConverterInterface
+     */
+    protected $schemaConverter;
+
+    /**
+     * @var \SprykerSdk\Acp\Schema\SchemaWriterInterface
+     */
+    protected $schemaWriter;
+
+    /**
+     * @param \SprykerSdk\Acp\Schema\SchemaConverterInterface $schemaConverter
+     * @param \SprykerSdk\Acp\Schema\SchemaWriterInterface $schemaWriter
+     */
+    public function __construct(SchemaConverterInterface $schemaConverter, SchemaWriterInterface $schemaWriter)
+    {
+        $this->schemaConverter = $schemaConverter;
+        $this->schemaWriter = $schemaWriter;
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return \Transfer\CreateDefaultEndpointsResponseTransfer
+     */
+    public function extendProjectSchema(CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer): CreateDefaultEndpointsResponseTransfer
+    {
+        $createDefaultEndpointsResponseTransfer = $this->addDefaultEndpointsToSchema($createDefaultEndpointsRequestTransfer);
+
+        $configurationSchemaGeneratedMessage = $this->convertConfigurationToSchema($createDefaultEndpointsRequestTransfer);
+
+        if ($configurationSchemaGeneratedMessage) {
+            $createDefaultEndpointsResponseTransfer->addError($configurationSchemaGeneratedMessage);
+        }
+
+        return $createDefaultEndpointsResponseTransfer;
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return \Transfer\CreateDefaultEndpointsResponseTransfer
+     */
+    protected function addDefaultEndpointsToSchema(
+        CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+    ): CreateDefaultEndpointsResponseTransfer {
+        $result = new CreateDefaultEndpointsResponseTransfer();
+
+        $schema = $this->getDefaultEndpointsSchema($createDefaultEndpointsRequestTransfer);
+
+        $process = new Process([
+            'vendor/bin/syncapi',
+            'schema:openapi:update',
+            '-f',
+            $createDefaultEndpointsRequestTransfer->getSchemaFile(),
+            $this->schemaWriter->writeSchemaToJsonString($schema),
```

Comment:
As we discussed, let's change the strategy to passing either an (absolute) file path or passing the contents of the YML file as-is. Converting back and forth between YML and JSON was a silly decision during the specification that I should've caught earlier.

Reply:
Will implement this on sync-api side first and come back to this topic afterwards